src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -85,7 +94,33 @@ public void Dispose()
 
     public string MaskSecrets(string input)
     {
-        return _secretMasker.MaskSecrets(input);
+        return _secretMasker.MaskSecrets(input, _detectionAction);
+    }
+
+    private void ProcessDetection(Detection detection)
+    {
+        if (_detectionsForTelemetry.Count < _maxDetectionsForTelemetry &&
```

Comment:
looks like we're adding even if knob is not enabled

Reply:
This was deliberate based on earlier discussion, the gathering is done irrespective of flag, but I think I can have the job start enable collection based on the knob. I'll do that.

Reply:
To clarify, as is, we don't *send* the telemetry without the knob but we do gather it in memory. And I will fix it to do neither without the knob.